sys/riotboot/reset.c

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2020 Benjamin Valentin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     sys_riotboot_serial
+ * @{
+ *
+ * @file
+ * @brief       Trigger reset to riotboot.
+ *
+ * @author      Benjamin Valentin <benpicco@googlemail.com>
+ * @}
+ */
+
+#include <string.h>
+#include "cpu.h"
+#include "riotboot/hdr.h"
+
+__attribute__((weak))
+void usb_board_reset_in_bootloader(void)
+{
+    uint32_t *magic = (void*)RIOTBOOT_MAGIC_ADDR;
+
+    irq_disable();
+    *magic = RIOTBOOT_MAGIC;
+    NVIC_SystemReset();
```

Comment:
This code is made to be architecture agnostic, so better use `pm_reboot` here

Reply:
I'm just a bit afraid that this might mess with the magic address - some arch code does quit a bit of cleanup in `pm_reboot()`.